### PR TITLE
Add Linux prep instructions to developer guide

### PR DESCRIPTION
**Pull request by renxida**
[skip ci]

There are some apt packages needed before it would work.

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -3,6 +3,14 @@
 Each sub-project has its own developer guide. If you would like to work across
 projects, these instructions should help you get started:
 
+
+### Prepare your system
+
+(these instructions are tested on Ubuntu 24.04 which has clang-18 and python3.12 by default)
+
+```
+sudo apt update && sudo apt install -y python-is-python3 python3.12-venv python3-dev clang lld
```
Comment (ScottTodd) [inline docs/developer_guide.md:12]: Might want to recommend https://github.com/pyenv/pyenv over `python-is-python3 python3.12-venv python3-dev`

I regularly switch between 3.10, 3.11, 3.12, 3.13, and 3.13t when testing packages.

Comment (ScottTodd) [reply docs/developer_guide.md:12]: That's fine, but I wouldn't over-index on Ubuntu 24.04 and Python 3.12.

Comment (stbaione) [reply docs/developer_guide.md:12]: Personally, I like just linking to pyenv and saying it's a good tool to use for managing multiple python environments. Probably too verbose for our docs, but if you wanted to check it out, here are some setup instructions:

## Pre-Reqs for ubuntu from their [wiki](https://github.com/pyenv/pyenv/wiki)
```bash
sudo apt update; sudo apt install build-essential libssl-dev zlib1g-dev \
libbz2-dev libreadline-dev libsqlite3-dev curl git \
libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
```

## Installing pyenv from instructions [here](https://github.com/pyenv/pyenv?tab=readme-ov-file#installation)
```bash
curl https://pyenv.run | bash
```

## Add to .bashrc
It's also outputted by installer after running installation:
```bash
echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bashrc
echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc
echo 'eval "$(pyenv init -)"' >> ~/.bashrc
```

## Run source ~/.bashrc
```bash
source ~/.bashrc
```

## Check installation
```bash
pyenv --version
```

## Install whichever version you need (i.e. for 3.12)
```bash
pyenv install 3.12
```

## Set as local version
In your workspace:
```bash
pyenv local 3.12
```

## Check python version
```bash
python --version
```

Comment (renxida) [reply docs/developer_guide.md:12]: yup i saw those. Made this based on that:

https://github.com/nod-ai/shark-ai/pull/585

Turns out pyenv can be just an extra step you do to replace python3-venv

I think we should just do venv for simplicity's sake but i'll add a lil click-expand section on using pyenv

Comment (ScottTodd) [reply docs/developer_guide.md:12]: > Turns out pyenv can be just an extra step you do to replace python3-venv
> 
> I think we should just do venv for simplicity's sake but i'll add a lil click-expand section on using pyenv

pyenv and venv are different. Prior discussion in this repo: https://github.com/nod-ai/shark-ai/pull/524#discussion_r1842970285

File: docs/developer_guide.md [hunk 2]
```diff
@@ -3,6 +3,14 @@
 Each sub-project has its own developer guide. If you would like to work across
 projects, these instructions should help you get started:
 
+
+### Prepare your system
+
+(these instructions are tested on Ubuntu 24.04 which has clang-18 and python3.12 by default)
```
[the remark this answers is not in the document]
Comment (renxida) [reply docs/developer_guide.md]: good point. fixed

File: docs/developer_guide.md (hunk 1 above)
Comment (renxida) [reply docs/developer_guide.md]: Sorry. mybad. not replace, but supplement.

Repalces python-is-python3, and docs adjusted accordingly.

**Comment by renxida** [comment]
Edits incorporated & pls re-review.

File: docs/developer_guide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (renxida) [reply docs/developer_guide.md]: Fixed

File: docs/developer_guide.md [hunk 3]
```diff
@@ -3,6 +3,55 @@
...
+command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"
+eval "$(pyenv init -)"
+```
+
+Finally, install a pyenv-managed version of python
+
+```bash
+pyenv install 3.12 # or whichever python version you'd like
+pyenv local 3.12
+```
```
Comment (ScottTodd) [inline docs/developer_guide.md:49]: At least link to the upstream documentation here. I wouldn't even include the inlined copy of their docs, but that's up to you. Each line of documentation we add here is an extra line of documentation to keep from going stale. Direct people to upstream, which will actually be maintained.

File: docs/developer_guide.md [hunk 4]
```diff
@@ -3,6 +3,55 @@
...
+sudo apt update && sudo apt install -y clang lld
+```
+
+### Prepare your python environment
+
+Install:
+
+```
+python-is-python3 python3-venv python3-dev
+```
```
Comment (ScottTodd) [inline docs/developer_guide.md:20]: Prefix these with `sudo apt install`

Also nit: set the syntax for code blocks (\`\`\`bash)

**Comment by renxida** [comment]
ack merged before i saw Chris's comment. Let me incorporate the comments in another PR,

**Pull request merged**